feat: add base URL support for bracket links

weakish opened the pull request:
Close #9

weakish committed:
feat: add base URL support for bracket links

All the code is written by Claude Sonnet 4
with the following prompt:

Add a baseUrl option, e.g. if baseUrl is /foo, then [[link]] will be linked to /foo/link/ instead of /link/.

weakish committed:
chore: ignore package-lock.json

weakish committed:
test: simplify baseUrl tests

weakish committed:
refactor: simplify base URL concatenation in bracket link parsing

weakish committed:
docs: simplify baseUrl option doc

johackim commented on .gitignore:
```diff
@@ -137,3 +137,6 @@ dist
 .svelte-kit
+
+# package lock files
+package-lock.json
```
Can you remove this?

johackim commented on README.md:
```diff
@@ -49,14 +49,18 @@ import rehypeStringify from 'rehype-stringify';
 const { value } = unified()
     .use(remarkParse)
-    .use(remarkObsidian)
+    .use(remarkObsidian, { baseUrl: '/docs' })
```
Can you remove this?

johackim commented on README.md:
```diff
@@ -49,14 +49,18 @@ import rehypeStringify from 'rehype-stringify';
     .processSync('[[Hello world]]');
 
-console.log(value); // <a href="/hello-world">Hello world</a>
+console.log(value); // <a href="/docs/hello-world">Hello world</a>
```
Can you remove this?

johackim commented on __tests__/index.spec.js:
```diff
@@ -236,3 +245,20 @@ test.skip('Should support ==highlight **bold text**==', async () => {
+    const output = String(await remark().use(plugin).process(text));
+
+    expect(output).toContain('<a href="/internal-link" title="Internal link">Internal link</a>');
+});
```
Can you remove this?

johackim reviewed (commented):
Thanks for you PR, can you make theses changes?

weakish committed:
test: remove unnecessary tests

Also remove the baseUrl option in usage sample
to keep the sample with remark and with unified consistent.

Thank @johackim for reviewing :-)

johackim commented on .gitignore:
```diff
@@ -136,4 +136,4 @@ dist
 
 # SvelteKit build / generate output
-.svelte-kit
+.svelte-kit
```
Can you remove this change with a command like `awk -i inplace '1' .gitignore`?

weakish replied:
Sorry, I reverted accidentally removal of new line at the end of the file (via `git checkout master .gitignore`) in [3632a5b](https://github.com/johackim/remark-obsidian/pull/10/commits/3632a5b8e08bb3ebb36241ff14111969a85ccc6e)

johackim reviewed (commented):
Thanks @weakish for your changes!

A last change to make and I merge your PR 😀

weakish committed:
fix: ensure newline at end of .gitignore file

johackim commented:
:tada: This PR is included in version 1.9.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/remark-obsidian/v/1.9.0)
- [GitHub release](https://github.com/johackim/remark-obsidian/releases/tag/v1.9.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
